lib/connection.cpp

```diff
@@ -2150,9 +2152,9 @@ UnorderedMap<std::pair<QString, QString>, QOlmInboundGroupSessionPtr> Connection
     return database()->loadMegolmSessions(room->id(), picklingMode());
 }
 
-void Connection::saveMegolmSession(Room* room, const QString& senderKey, QOlmInboundGroupSession* session)
+void Connection::saveMegolmSession(Room* room, const QString& senderKey, QOlmInboundGroupSession* session, const QString& ed25519Key)
 {
-    database()->saveMegolmSession(room->id(), senderKey, session->sessionId(), session->pickle(picklingMode()));
+    database()->saveMegolmSession(room->id(), senderKey, session->sessionId(), ed25519Key, session->pickle(picklingMode()));
```

Comment:
What if `ed25519Key` comes empty? Either a validation in either of the two `saveMegolmSession()` methods, or a database constraint (and a check that the query completed successfully) would be worthwhile. Or is there already a check that I missed?

Reply:
done